pgjdbc/src/main/java/org/postgresql/core/v3/SimpleParameterList.java

```diff
@@ -446,6 +496,35 @@ public void clear() {
     return paramValues;
   }
 
+  @Override
+  public int getIndex(String parameterName) throws PSQLException {
+    if (this.paramNameIndex == null) {
+      throw new PSQLException(
+          GT.tr("The ParameterList was not created with named parameters."),
+          PSQLState.INVALID_PARAMETER_VALUE);
+    }
+
+    final Integer index = this.paramNameIndex.get(parameterName);
+
+    if (index == null) {
+      throw new PSQLException(
+          GT.tr("The parameterName was not found : {0}. The following names are known : \n\t {1}",
+              parameterName, this.paramNameIndex.keySet()), PSQLState.INVALID_PARAMETER_VALUE);
+    }
+
+    return index;
+  }
+
+  @Override
+  public List<String> getParameterNames() throws PSQLException {
+    if (this.paramNames == null) {
+      throw new PSQLException(
+          GT.tr("The ParameterList was not created with named parameters."),
+          PSQLState.INVALID_PARAMETER_VALUE);
+    }
+    return this.paramNames;
```

Comment:
should this be an unmodifiable wrapper? (Collections.unmodifiableList)

Reply:
Yes, I like that. Then we don't need to copy anything.